subxt/src/tx/tx_client.rs

```diff
@@ -452,50 +455,323 @@ where
+        self.validate_at(latest_block_ref).await
+    }
+
+    /// Validate a transaction by submitting it to the relevant Runtime API. A transaction that is
+    /// valid can be added to a block, but may still end up in an error state.
+    ///
+    /// Returns `Ok` with a [`ValidationResult`], which is the result of attempting to dry run the extrinsic.
+    pub async fn validate_at(
+        &self,
+        at: impl Into<BlockRef<T::Hash>>,
+    ) -> Result<ValidationResult, Error> {
+        let block_hash = at.into().hash();
+
+        // Approach taken from https://github.com/paritytech/json-rpc-interface-spec/issues/55.
+        let mut params = Vec::with_capacity(8 + self.encoded.0.len() + 8);
+        2u8.encode_to(&mut params);
+        params.extend(self.encoded().iter());
+        block_hash.encode_to(&mut params);
+
+        let res: Vec<u8> = self
+            .client
+            .backend()
+            .call(
+                "TaggedTransactionQueue_validate_transaction",
+                Some(&params),
+                block_hash,
+            )
+            .await?;
+
+        ValidationResult::try_from_bytes(res)
     }
 
     /// This returns an estimate for what the extrinsic is expected to cost to execute, less any tips.
     /// The actual amount paid can vary from block to block based on node traffic and other factors.
     pub async fn partial_fee_estimate(&self) -> Result<u128, Error> {
         let mut params = self.encoded().to_vec();
         (self.encoded().len() as u32).encode_to(&mut params);
+        let latest_block_ref = self.client.backend().latest_best_block_ref().await?;
+
         // destructuring RuntimeDispatchInfo, see type information <https://paritytech.github.io/substrate/master/pallet_transaction_payment_rpc_runtime_api/struct.RuntimeDispatchInfo.html>
         // data layout: {weight_ref_time: Compact<u64>, weight_proof_size: Compact<u64>, class: u8, partial_fee: u128}
         let (_, _, _, partial_fee) = self
             .client
-            .rpc()
-            .state_call::<(Compact<u64>, Compact<u64>, u8, u128)>(
+            .backend()
+            .call_decoding::<(Compact<u64>, Compact<u64>, u8, u128)>(
                 "TransactionPaymentApi_query_info",
                 Some(&params),
-                None,
+                latest_block_ref.hash(),
             )
             .await?;
         Ok(partial_fee)
     }
 }
+
+impl ValidationResult {
+    fn try_from_bytes(bytes: Vec<u8>) -> Result<ValidationResult, crate::Error> {
+        // TaggedTransactionQueue_validate_transaction returns this:
+        // https://github.com/paritytech/substrate/blob/0cdf7029017b70b7c83c21a4dc0aa1020e7914f6/primitives/runtime/src/transaction_validity.rs#L210
+        // We copy some of the inner types and put the three states (valid, invalid, unknown) into one enum,
+        // because from our perspective, the call was successful regardless.
+        if bytes[0] == 0 {
```

Comment:
nit: In case we somehow don't receive an `Result<ValidTransaction, TransactionValidityError>` and end-up with some empty bytes, would it be beneficial to do a `bytes.get(0).expect("Invalid wire format for ValidationResult")`?